### PR TITLE
fix/premium-username-flow

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Removes logic that no longer applies with current subscription model
- return monthly subscription for premium username

Fixes # (issue)

## Loom
https://www.loom.com/share/8a71b0d65334440daad8a0521b33dc33

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] Setup stripe environments variables
- [ ] Edit some constants to emulate local development as production and not self host.
- [ ] Go to settings/profile and pick a premium username
- [ ] Complete payment

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/getting-started/steps-views/UserSettings.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ const UserSettings = (props: IUserSettingsProps) => {
     onSuccess: onSuccess,
   });
   const { data: stripeCustomer } = trpc.viewer.stripeCustomer.useQuery();
-  const paymentRequired = stripeCustomer?.isPremium ? !stripeCustomer?.paidForPremium : false;
+  const paymentRequired = stripeCustomer?.isPremium;
```
Comment (alannnc) [inline apps/web/components/getting-started/steps-views/UserSettings.tsx:45]: No longer needed for one time payment

File: apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx [hunk 2]
```diff
@@ -106,7 +97,7 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
         setIsInputUsernamePremium(data.premium);
         setUsernameIsAvailable(data.available);
       }, 150),
-    []
+    [currentUsername]
```
Comment (alannnc) [inline apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx:100]: This param was missing and linter warning was showing

File: packages/app-store/stripepayment/api/subscription.ts [hunk 3]
```diff
@@ -49,12 +36,20 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       return;
     }
 
-    const isCurrentlyPremium = hasKeyInMetadata(userData, "isPremium") && !!userData.metadata.isPremium;
-
     const return_url = `${process.env.NEXT_PUBLIC_WEBAPP_URL}/api/integrations/stripepayment/paymentCallback?checkoutSessionId={CHECKOUT_SESSION_ID}&callbackUrl=${callbackUrl}`;
-    const createSessionParams: Stripe.BillingPortal.SessionCreateParams = {
+    const createSessionParams: Stripe.Checkout.SessionCreateParams = {
```
Comment (alannnc) [inline packages/app-store/stripepayment/api/subscription.ts:40]: Changing billing to checkout

File: packages/app-store/stripepayment/api/subscription.ts [hunk 4]
```diff
@@ -72,87 +67,21 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-          customProductsSession.push({ prices: [PRO_PLAN_PRICE], product: PRO_PLAN_PRODUCT_ID });
-        }
-      } else if (action === UsernameChangeStatusEnum.DOWNGRADE) {
-        actionText = "Downgrade your plan account";
-        if (isCurrentlyPremium) {
-          customProductsSession.push({ prices: [PRO_PLAN_PRICE], product: PRO_PLAN_PRODUCT_ID });
-        }
-      }
-
-      const configuration = await stripe.billingPortal.configurations.create({
```
Comment (alannnc) [inline packages/app-store/stripepayment/api/subscription.ts:125]: Remove upgrade/downgrade logic

File: packages/app-store/stripepayment/lib/constants.ts [hunk 5]
```diff
@@ -1,8 +1,3 @@
...
-export const PREMIUM_PLAN_PRICE = process.env.NEXT_PUBLIC_STRIPE_PREMIUM_PLAN_PRICE || "";
-export const IS_PREMIUM_NEW_PLAN = process.env.NEXT_PUBLIC_IS_PREMIUM_NEW_PLAN === "1" ? true : false;
-export const PREMIUM_NEW_PLAN_PRICE = process.env.NEXT_PUBLIC_STRIPE_PREMIUM_NEW_PLAN_PRICE || "";
-export const PRO_PLAN_PRICE = process.env.NEXT_PUBLIC_STRIPE_PRO_PLAN_PRICE || "";
-export const FREE_PLAN_PRODUCT_ID = process.env.STRIPE_FREE_PLAN_PRODUCT_ID || "";
-export const PRO_PLAN_PRODUCT_ID = process.env.STRIPE_PRO_PLAN_PRODUCT_ID || "";
-export const PREMIUM_PLAN_PRODUCT_ID = process.env.STRIPE_PREMIUM_PLAN_PRODUCT_ID || "";
+export const PREMIUM_MONTHLY_PLAN_PRICE = process.env.NEXT_PUBLIC_STRIPE_PREMIUM_PLAN_PRICE_MONTHLY || "";
+export const PREMIUM_PLAN_PRODUCT_ID = process.env.NEXT_PUBLIC_STRIPE_PREMIUM_PLAN_PRODUCT_ID || "";
+export const STRIPE_TEAM_MONTHLY_PRICE_ID = process.env.NEXT_PUBLIC_STRIPE_TEAM_MONTHLY_PRICE_ID || "";
```
Comment (alannnc) [inline packages/app-store/stripepayment/lib/constants.ts:3]: Leaving only team billing and premium products or prices.

File: packages/app-store/stripepayment/lib/getCustomerAndCheckoutSession.ts [hunk 6]
```diff
@@ -6,19 +6,19 @@ export async function getCustomerAndCheckoutSession(checkoutSessionId: string) {
...
   if (!customerOrCustomerId) {
-    return { checkoutSession, customer: null };
+    return { checkoutSession, stripeCustomer: null };
   }
 
   if (typeof customerOrCustomerId === "string") {
     customerId = customerOrCustomerId;
   } else if (customerOrCustomerId.deleted) {
-    return { checkoutSession, customer: null };
+    return { checkoutSession, stripeCustomer: null };
```
Comment (alannnc) [inline packages/app-store/stripepayment/lib/getCustomerAndCheckoutSession.ts:15]: return same typed response

File: packages/trpc/server/routers/viewer.tsx [hunk 7]
```diff
@@ -582,23 +583,35 @@ const loggedInViewerRouter = router({
         throw new TRPCError({ code: "NOT_FOUND", message: "User not found" });
       }
       const metadata = userMetadata.parse(userToUpdate.metadata);
-      // Checking the status of payment directly from stripe allows to avoid the situation where the user has got the refund or maybe something else happened asyncly at stripe but our DB thinks it's still paid for
-      // TODO: Test the case where one time payment is refunded.
-      const premiumUsernameCheckoutSessionId = metadata?.checkoutSessionId;
+
+      const isPremium = metadata?.isPremium;
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer.tsx:587]: This was and should be our only flag going forward.

File: apps/web/playwright/change-username.e2e.ts [hunk 8]
```diff
@@ -98,11 +82,11 @@ test.describe("Change username on settings", () => {
     await expect(page).toHaveURL(/.*checkout.stripe.com/);
   });
 
-  test("User PRO can update to PREMIUM username", async ({ page, users }, testInfo) => {
+  test("User can update to PREMIUM username", async ({ page, users }, testInfo) => {
     // eslint-disable-next-line playwright/no-skipped-test
     test.skip(!IS_STRIPE_ENABLED, "It should only run if Stripe is installed");
     test.skip(IS_SELF_HOSTED, "It shouldn't run on self hosted");
-    const user = await users.create({ plan: UserPlan.PRO });
+    const user = await users.create();
```
Comment (alannnc) [inline apps/web/playwright/change-username.e2e.ts:89]: Plan logic it's gone

**Comment by zomars** [comment]
Can we get a loom? @alannnc

**Comment by alannnc** [comment]
> Can we get a loom? @alannnc

Loom added! @zomars

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**